grails-app/controllers/org/pih/warehouse/product/ProductSupplierController.groovy

```diff
@@ -27,7 +27,29 @@ class ProductSupplierController {
 
     def list = {
         params.max = Math.min(params.max ? params.int('max') : 10, 100)
-        [productSupplierInstanceList: ProductSupplier.list(params), productSupplierInstanceTotal: ProductSupplier.count()]
+        def productSuppliers = ProductSupplier.createCriteria().list(max: params.max, offset: params.offset) {
+            if (params.product?.id) {
+                eq('product.id', params.product.id)
+            }
+            if (params.supplierId) {
+                eq('supplier.id', params.supplierId)
+            }
+            if (params.manufacturerId) {
+                eq('manufacturer.id', params.manufacturerId)
+            }
+            if (params.q) {
+                or {
+                    ilike("productCode", "%" + params.q + "%")
+                    ilike("code", "%" + params.q + "%")
```

Comment:
We probably need to match against Product properties too but I'll leave that up to Kelsey. See my comments in the ticket.